app-modules/service-management/tests/ServiceRequest/CreateServiceRequestTest.php

```diff
@@ -464,11 +467,93 @@
         ])
         ->create();
 
-    $serviceRequest = ServiceRequest::factory()->create([
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
         'priority_id' => ServiceRequestPriority::factory()->create([
             'type_id' => $serviceRequestTypesWithManager->getKey(),
         ])->getKey(),
-    ]);
+    ]));
+
+    livewire(CreateServiceRequest::class)
+        ->fillForm($request->toArray())
+        ->fillForm([
+            'respondent_id' => Contact::factory()->create()->getKey(),
+        ])
+        ->call('create');
+
+    $serviceRequest = ServiceRequest::first();
 
     expect($serviceRequest->assignments()->first())->user->id->toBe($user->getKey());
 });
+
+test('check round robin assignment', function () {
+    User::factory()->licensed(LicenseType::cases())->create();
+    $team = Team::factory()->create();
+
+    $serviceRequestTypesWithManager = ServiceRequestType::factory()
+        ->hasAttached(
+            factory: $team,
+            relationship: 'managers'
+        )
+        ->state([
+            'assignment_type' => ServiceRequestTypeAssignmentTypes::RoundRobin,
+        ])
+        ->create();
+
+    $request = collect(CreateServiceRequestRequestFactory::new()->create([
+        'priority_id' => ServiceRequestPriority::factory()->create([
+            'type_id' => $serviceRequestTypesWithManager->getKey(),
+        ])->getKey(),
+    ]));
+    $users = User::orderBy('name')->orderBy('id')->get();
+
+    foreach ($users as $user) {
+        $user->grantLicense(LicenseType::RecruitmentCrm);
+        $user->teams()->attach($team);
+        $user->givePermissionTo('service_request.view-any');
+        $user->givePermissionTo('service_request.create');
+        $user->refresh();
+        actingAs($user);
+
+        livewire(CreateServiceRequest::class)
+            ->assertSuccessful()
+            ->fillForm($request->toArray())
+            ->call('create');
+
+        $serviceRequestDataObject = new ServiceRequestDataObject(
+            division_id: $request['division_id'],
+            status_id: $request['status_id'],
+            type_id: $request['type_id'],
+            priority_id: $request['priority_id'],
+            title: $request['title'],
+            close_details: $request['close_details'],
+            res_details: $request['res_details'],
+            respondent_type: $request['respondent_type'],
+            respondent_id: $request['respondent_id'],
+        );
+
+        app(CreateServiceRequestAction::class)->execute($serviceRequestDataObject);
```

Comment:
I am not sure why you are calling the component `create` bother here and then also executing the Action. The Action should be trigger in the component.